docs/shared/checkpoint_learn_serialize.md

```diff
@@ -0,0 +1,118 @@
+\page ckpt_learn_serialize How to Serialize Data
+
+*Serialization* is the process of converting an object into a simple format
+that can be stored or transmitted and reconstructed later.
+checkpoint translates the object into a set of contiguous bits
+and provides the steps to reverse the process, i.e. to reconstitute
+the object from the set of bits.
+
+The extraction of the data from a series of bytes is called
```

Comment:
Please use consistent terminology - "set of bits" vs "series of bytes". I prefer the latter.